packages/firestore/test/unit/specs/limbo_spec.test.ts

```diff
@@ -287,4 +288,49 @@ describeSpec('Limbo Documents:', [], () => {
         .expectEvents(query, { removed: [doc2] });
     }
   );
+
+  specTest(
+    'Limbo documents are resolved after primary tab failover',
+    ['multi-client'],
+    () => {
+      const query = Query.atPath(path('collection'));
+      const doc1 = doc('collection/a', 1000, { key: 'a' });
+      const doc2 = doc('collection/b', 1001, { key: 'b' });
+      const deletedDoc2 = deletedDoc('collection/b', 1005);
+
+      return (
+        client(0, false)
+          .becomeVisible()
+          .client(1)
+          .userListens(query)
+          .expectEvents(query, { fromCache: true })
+          .client(0)
+          .expectListen(query)
+          .watchAcksFull(query, 1002, doc1, doc2)
+          .client(1)
+          .expectEvents(query, { added: [doc1, doc2] })
+          .client(0)
+          .watchRemovesDoc(doc2.key, query)
+          .watchSnapshots(1003)
+          .expectLimboDocs(doc2.key)
+          .shutdown()
+          .client(1)
+          .expectEvents(query, { fromCache: true })
```

Comment:
Hrm...  Not 100% sure if this is expected / desirable.  I think ideally we'd pick a new primary and let it try to connect to the backend before triggering fromCache: true in all the secondary tabs.  Perhaps add a TODO(multi-tab)?

Reply:
We raise `fromCache` since the query is in limbo.